zerver/views/streams.py

```diff
@@ -587,6 +588,45 @@ def add_subscriptions_backend(
     return json_success(result)
 
 
+def get_stream_policy(stream: Stream) -> str:
+    if (
+        stream.invite_only is False
+        and stream.history_public_to_subscribers is True
+        and stream.is_web_public is True
+    ):
+        return gettext_lazy("Web-public")
+
+    if (
+        stream.invite_only is False
+        and stream.history_public_to_subscribers is True
+        and stream.is_web_public is False
+    ):
+        return gettext_lazy("Public")
+
+    if (
+        stream.invite_only is True
+        and stream.history_public_to_subscribers is True
+        and stream.is_web_public is False
+    ):
+        return gettext_lazy("Private, shared history")
+
+    if (
+        stream.invite_only is True
+        and stream.history_public_to_subscribers is False
+        and stream.is_web_public is False
+    ):
+        return gettext_lazy("Private, protected history")
+
+    if (
+        stream.invite_only is False
+        and stream.history_public_to_subscribers is False
+        and stream.is_web_public is False
+    ):
+        return gettext_lazy("Public, protected history")
+
+    return ""
```

Comment:
Is it possible to avoid duplicating this with the existing code we have for displaying notifications for changes to these settings?  We really don't want to have multiple copies of this logic to test and have bugs in.

Reply:
@timabbott I have made suggested changes, please review.